python/cppyy/numba_ext.py

```diff
@@ -130,14 +160,57 @@ def get_call_type(self, context, args, kwds):
         except KeyError:
             pass
 
-        ol = CppFunctionNumbaType(self._func.__overload__(*(numba2cpp(x) for x in args)), self._is_method)
+        args_cpp = []
+        for i, arg in enumerate(list(args)):
+            # If the user explicitly passes an argument using numba CPointer, the regex match is used
+            # to detect the pass by reference since the dispatcher always returns typeref[val*]
+            match = re.search(r"typeref\[(.*?)\*\]", str(arg))
+            if match:
+                literal_val = match.group(1)
+                arg_type = numba.typeof(eval(literal_val))
+                args_cpp.append(to_ref(numba2cpp(arg_type)))
+            else:
+                args_cpp.append(numba2cpp(arg))
+        args_combinations = itertools.product(*args_cpp)
+
+        for arg_combo in args_combinations:
```

Comment:
This should be part of the updates Baidyanath made to CPyCppyy argument matching and not live in Python (it's all rather slow in practice, so having it Python-side won't help).